cmd/soroban-cli/src/config/secret.rs

```diff
@@ -23,52 +21,34 @@ pub enum Error {
     SeedPhrase(#[from] sep5::error::Error),
     #[error(transparent)]
     Ed25519(#[from] ed25519_dalek::SignatureError),
-    #[error("Invalid address {0}")]
-    InvalidAddress(String),
+    #[error("cannot parse secret (S) or seed phrase (12 or 24 word)")]
+    InvalidSecretOrSeedPhrase,
     #[error(transparent)]
     Signer(#[from] signer::Error),
 }
 
 #[derive(Debug, clap::Args, Clone)]
 #[group(skip)]
 pub struct Args {
-    /// Add using `secret_key`
+    /// (deprecated) Add secret (S) key
```

Comment:
Ah so we will just have a `secret` in the future? If so could we add it here? It seems weird to deprecate without an alternative.

Reply:
I think we can accept both as inputs without using any options. Wdyt?

Reply:
This change seems like it simplifies things!

I wonder if a bit more explanation in the doc comments & deprecations would help to let the user know that they will now be prompted to enter their key/passphrase. I was a bit confused at first how this would work, until I tried it out. 

Something like this maybe?
```suggestion
    /// (deprecated) Enter secret (S) key when prompted
```

Reply:
I like this change, but there should be a clear guidance of using `SOROBAN_SECRET_KEY` in help.
```
 > stellar keys generate --help
Generate a new identity with a seed phrase, currently 12 words
```

Reply:
> > stellar keys generate --help
> Generate a new identity with a seed phrase, currently 12 words

- https://github.com/stellar/stellar-cli/issues/1816